src/main/resources/data/minecoprocessors/books/manual.txt

```diff
@@ -1,130 +1,291 @@
+
+In most cases, binary or hexadecimal representations are best, as we are often dealing with the bits in registers.
+~~~
+§nCore Registers§r
+
+The CPU has four one byte registers §1a, b, c,§r and §1d§r to store values and calculate.
+
+These "variables" do not change any of the signals outside the CPU - they are only internal.
 ~~~
-§1Instructions§r
+§nPorts and Port Config§r
 
-§lMOV a, b§r Move
-§lPUSH a§r Push to Stack
-§lPOP a§r Pop from Stack
-§lNOP§r No Operation
-§lHLT§r Halt Processor
-§lCLC§r Clear Carry
-§lSEC§r Set Carry
-§lCLZ§r Clear Zero
-§lSEZ§r Set Zero
-§lHLT§r Halt Processor
-§lWFE§r Sleep (experimental)
+The CPU interacts with the world around via special registers. These are mapped to the four sides of the device, and we need to tell the CPU if we like to read signals from a side, or want to output redstone power there.
+
+Additionally, we can
 ~~~
-§1Arithmetic Instructions§r
+specify for each I/O if we are interested in the numeric "strength" value of the signal, or only the on/off data.
+§oAlso known as §1analog§r and §1digital§r mode.§r
+
+Lastly, ports can also be used to reset the controller by leaving a side unconfigured or "unconfiguring" it explicitly.
 
-§lADD a, b§r Add
-§lSUB a, b§r Subtract
-§lMUL a§r Multiply
-§lDIV a§r Divide
-§lINC a§r Increment by 1
-§lDEC a§r Decrement by 1
 ~~~
-§1Logical Instructions§r
+Signals are accessible primarily through the registers §1pf pb pl pr§r:
+
+§80001b | §1pf§r : front
+§80010b | §1pb§r : back
+§80100b | §1pl§r  : left
+§81000b | §1pr§r : right
 
-§lAND a, b§r Bitwise AND
-§lOR a, b§r Bitwise OR
-§lXOR a, b§r Bitwise XOR
-§lNOT a§r Bitwise NOT
+Gray annotated are the binary bit values to configure the I/Os in the §1ports§r and §1adc§r registers.
 ~~~
-§1Shift Instructions§r
+The first instruction in a listing is normally to write the register §1ports§r, where cleared bit configure outputs and set bits inputs. E.g.
 
-§lSHL a§r Shift Left
-§lSHR a§r Shift Right
-§lSAR a§r Arithmetic Right
-§lSAL a§r Arithmetic Left
-§lROR a§r Rotate Right
-§lROL a§r Rotate Right
+  §1MOV ports, 1010b§r
+
+defines §1pf§r and §1pl§r as outputs, §1pb§r and §1pr§r as inputs. To use an I/O for reset, set its bit
 ~~~
-§1Loop Instructions§r
+in the low and high nibble:
+
+  §1MOV ports, 0x33§r
+
+configures §1pf§r and §1pb§r as reset, §1pl§r and §1pr§r as output.
 
-§lJMP label§r Jump
-§lJZ label§r Jump if Zero
-§lJNZ label§r not Zero
-§lJC label§r if Carry
-§lJNC label§r if no Carry
-§lDJNZ a, label§r Dec and Jump if not zero
-§lCALL label§r Call
-§lRET§r Return
-§lCMP a, b§r Compare
+By default I/Os are in digital mode. Hence, we read the value §10x00§r if an input signal is §40§r, and §10x01§r for any
 ~~~
-§1Number Formats§r
+strength from §41§r to §415§r.
+Reversely, writing §10§r to an output will result in an unpowered side, while any other value will yield strength §415§r.
```

Comment:
Consider changing this wording:
`Similarly, writing §10§r to an output port will yield strength §40§r, on that side, and any other value will yield signal strength §415§r.`

I admit that there might not be space on the page left for this longer line, but I'm not able to check that currently

Reply:
Typo of a comma, and maybe change to
`Similarly, writing §10§r to a digital output port will yield strength §40§r on that side, and any other value will yield signal strength §415§r.`